### PR TITLE
Support keyvault for secrets

### PR DESCRIPTION
**Pull request by carolynvs**
Credentials are resolved against Azure keyvault secrets

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Storage plugins allow Porter to store data, such as claims and credentials, in t
 The `azure.blob` plugin stores data in Azure Blob Storage. 
 
 1. Open, or create, `~/.porter/config.toml`.
-1. Add the following line to instruct Porter to use the azure blob storage plugin
+1. Add the following line activate the Azure blob storage plugin
```
Comment (vdice) [inline README.md]: Oops, missing a `to` betwixt `line` and `activate`.

File: README.md [hunk 2]
```diff
@@ -37,6 +37,36 @@ The `azure.blob` plugin stores data in Azure Blob Storage.
 1. [Copy the connection string][connstring] for the storage account. Then set it as an environment variable named 
     `AZURE_STORAGE_CONNECTION_STRING`.
 
+## Secrets
+
+Secrets plugins allow Porter to inject secrets into credential sets.
+
+For example, your team may have a shared key vault with a database password, you can use the keyvault plugin
```
Comment (vdice) [inline README.md]: nit: sentence reads slightly awkward.  Perhaps we can add an `if` here, for: `For example, if your team has a shared...`?

File: pkg/azure/azureconfig/config.go [hunk 3]
```diff
@@ -1,6 +1,17 @@
 package azureconfig
 
 type Config struct {
-	// Env is the environment variable from which the connection string should be loaded.
-	Env string
+
+	// EnvConnectionString is the environment variable from which the connection
+	// string should be loaded.
+	EnvConnectionString string `json:"env"`
```
Comment (vdice) [inline pkg/azure/azureconfig/config.go]: Do we want to create a new `Env` struct to house various fields (`ConnectionString`, `ServicePrincipalPrefix`, etc.?)  Would then be accessed via `cfg.Env.ConnectionString`, etc.

File: pkg/azure/keyvault/plugin.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	cnabsecrets "github.com/cnabio/cnab-go/secrets"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+)
+
+const PluginInterface = secrets.PluginInterface + ".azure.keyvault"
+
+var _ cnabsecrets.Store = &Plugin{}
+
+// Plugin is the plugin wrapper for storing claims in azure blob storage.
```
Comment (vdice) [inline pkg/azure/keyvault/plugin.go]: inaccurate comment here...

File: README.md [hunk 5]
```diff
@@ -37,6 +37,36 @@ The `azure.blob` plugin stores data in Azure Blob Storage.
...
+    [[secrets]]
+    name = "mysecrets"
+    plugin = "azure.keyvault"
+    
+    [secrets.config]
+    vault = "myvault"
+    ```
+1. [Create a key vault][keyvault] and set the vault name in the config with name of the vault.
+1. [Create a service principal][sp] and create an Access Policy on the vault giving Get and List secret permissions.
+1. Using credentials for the service principal set the environment variables: `AZURE_TENANT_ID`,`AZURE_CLIENT_ID`,  and `AZURE_CLIENT_SECRET`.
```
Comment (vdice) [inline README.md]: Do we want to add docs (here or elsewhere) around the use case of setting/using a service principal prefix?

File: README.md (hunk 1 above)
Comment (carolynvs) [reply README.md]: 100 internet points for using betwixt

File: pkg/azure/azureconfig/config.go (hunk 3 above)
Comment (carolynvs) [reply pkg/azure/azureconfig/config.go]: Let me think about this more and address improving the config in a follow-up PR. I'd like to streamline this as much as possible, right now the config struct is shared which is part of why things look awkward. 🤔

File: README.md (hunk 5 above)
Comment (carolynvs) [reply README.md]: I didn't want to document beyond the one case that I know we want to support. I'm still playing around with how to best deal with an auth solution that requires the least effort for users. So as part of that follow-up I mentioned, I'll document whatever we settle on.

**Comment by carolynvs** [comment]
Here is the follow up issue for configuration https://github.com/deislabs/porter-azure-plugins/issues/8

**Pull request merged**